libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/routefetch/RoutesAction.kt

```diff
@@ -13,4 +13,14 @@ sealed class RoutesAction : Action {
      * @param routes list of [NavigationRoute]
      */
     data class SetRoutes(val routes: List<NavigationRoute>) : RoutesAction()
+
+    /**
+     * The action is used to directly set the [NavigationRoute] supplied to NavigationView.
+     * @param routes list of [NavigationRoute]
+     * @param legIndex index of route leg
+     */
+    data class SetRoutesWithIndex(
+        val routes: List<NavigationRoute>,
+        val legIndex: Int
+    ) : RoutesAction()
```

Comment:
Not really important, but you could have added a parameter to existing `SetRoutes` with default value 0. 
Should we also add `legIndex` with a default value to `NavigationViewApi.startActiveGuidance(List<NavigationRoute>)`?

Reply:
Will add the index on a need basis.